Recovery KMS HCL stop working after upgrade from 1.0.12 to 1.1.0

### Terraform Version
1.3.2

### Provider version
1.1.0

### Affected Resource(s)
Provider itself

### Terraform Configuration Files

```hcl
provider "boundary" {
  addr             = var.boundary_address
  recovery_kms_hcl = <<EOT
kms "awskms" {
	purpose    = "recovery"
	key_id     = "global_recovery"
  region     = "${var.region}"
  access_key = "${var.aws_access_key}"
  secret_key = "${var.aws_secret_key}"
  kms_key_id = "${module.aws-kms.kms_boundary_recovery_key_id}"
}
EOT
}
```

### Actual Behavior

Terraform apply failed with the following error:

```text
Error: error calling read scope: error performing client request during Read call:
error generating recovery KMS workflow token: error encrypting recovery info: rpc error:
code = Canceled desc = grpc: the client connection is closing
│ 
│   with module.boundary.boundary_scope.global,
│   on boundary/scopes.tf line 2, in resource "boundary_scope" "global":
│    2: resource "boundary_scope" "global" {
```

I have the same error with Terraform 1.2.9 and boundary provider 1.1.0.
With the 1.0.12 it is working fine.

```
│ Error: error calling read scope: error performing client request during Read call: error generating recovery KMS workflow token: error encrypting recovery info: rpc error: code = Canceled desc = grpc: the client connection is closing
│ 
│   with boundary_scope.global,
│   on boundary.tf line 5, in resource "boundary_scope" "global":
│    5: resource "boundary_scope" "global" {


2022-10-11T11:50:04.669+0200 [ERROR] provider.terraform-provider-boundary_v1.1.0_x5:
  Response contains error diagnostic:
    @module=sdk.proto
    diagnostic_summary="error calling read scope: error performing client request during Read call: error generating recovery KMS workflow token: error encrypting recovery info: rpc error: code = Canceled desc = grpc: the client connection is closing"
    tf_proto_version=5.3
    tf_req_id=ea3bbaf7-d3cb-3f30-5708-a450c4df5127
    @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55
    diagnostic_detail=
    diagnostic_severity=ERROR
    tf_provider_addr=provider
    tf_resource_type=boundary_scope
    tf_rpc=ReadResource
    timestamp=2022-10-11T11:50:04.669+0200

2022-10-11T11:50:04.669+0200 [ERROR] vertex "boundary_scope.global" error:
  error calling read scope:
    error performing client request during Read call:
      error generating recovery KMS workflow token:
        error encrypting recovery info:
          rpc error:
            code = Canceled
            desc = grpc: the client connection is closing
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,14 @@
 
 Canonical reference for changes, improvements, and bugfixes for the Boundary Terraform provider.
 
+## 1.1.1 (October 12, 2022)
+
+### Bug Fixes
+
+* The plugin cleanup function is being called before the entire Terraform workflow is complete.
+  ([Issue](https://github.com/hashicorp/terraform-provider-boundary/issues/282)),
+  ([PR](https://github.com/hashicorp/terraform-provider-boundary/pull/285)).
+
 ## 1.1.0 (October 4, 2022)
 
 ### New and Improved

--- a/internal/provider/provider.go
+++ b/internal/provider/provider.go
@@ -131,7 +131,8 @@ func providerAuthenticate(ctx context.Context, d *schema.ResourceData, md *metaD
 			opts = append(opts, pluginutil.WithPluginExecutionDirectory(execDir.(string)))
 		}
 
-		wrapper, cleanUp, err := wrapper.GetWrapperFromHcl(
+		// TODO: cleanup plugin when finished
+		wrapper, _, err := wrapper.GetWrapperFromHcl(
 			ctx,
 			recoveryHclStr,
 			"recovery",
@@ -141,15 +142,6 @@ func providerAuthenticate(ctx context.Context, d *schema.ResourceData, md *metaD
 		}
 		if wrapper == nil {
 			return errors.New(`No "kms" block with purpose "recovery" found in "recovery_kms_hcl"`)
-		}
-		if cleanUp != nil {
-			// Terraform will cancel the context when work is complete.
-			go func() {
-				select {
-				case <-ctx.Done():
-					_ = cleanUp()
-				}
-			}()
 		}
 
 		md.recoveryKmsWrapper = wrapper

